accepted/00000-coco-attestation.md

```diff
@@ -0,0 +1,80 @@
+- Feature Name: Confidential Compute Attestation Report
+- Start Date: 2024-01-16
+
+# Summary
+[summary]: #summary
+
+SUSE Manager / Uyuni should help with Confidential Compute Attestation and being able to generate a report page for systems which should run in this mode.
+
+# Motivation
+[motivation]: #motivation
+
+More and more systems run as virtual machines or (public) cloud instances.
+Customers want to be sure that data of these VMs cannot be read by administrators or the cloud provider.
+Confidential Compute is the way to solve this problem.
+But that a VM run in confidential mode needs to be attested and checked on regular base.
+
+SUSE Manager should provide an attestation report for a system.
+Also a history of the past checks should be kept and shown.
+
+The attestation is a process of multiple checks.
+SUSE Manager should provide a framework for executing such checks, store the results in the database and show the latest and historic results.
+The number of checks should be easy extendable in future versions.
+
+
+# Detailed design
+[design]: #detailed-design
+
+> This is the bulk of the RFC. Explain the design in enough detail for somebody familiar with the product to understand, and for somebody familiar with the internals to implement.
+>
+> This section should cover architecture aspects and the rationale behind disruptive technical decisions (when applicable), as well as corner-cases and warnings. Whenever the new feature creates new user interactions, this section should include examples of how the feature will be used.
+
+
+
+### Ideas around SNP attestation
+
+https://github.com/virtee/snpguest/blob/main/docs/regular.PNG
+
+- Apply a salt state on the VM
+  - send a nonce file
+  - request the attestation report
+  - request the vcek certificate
+- Store the data into the database
+  - Status: pending
+  - Store nonce, report and vcek data
+  - assign to a systemid and an execution date
+- Build a Container which does the attestation
```

Comment:
What about having an .sls for this task, so we can perform these salt states on each VM, instead of building a container to run the attestation? Also, this approach will facilitate scheduling it as an action every X time, and it will help to run it in parallel for all the minions we want to check it.

Reply:
The attestation container is for more security. If you attest things on a hacked VM the result cannot be trusted.
The verification should happen on a different system. A container sounds like a perfect candidate for this job.